templates/site/js/main.js

```diff
@@ -114,129 +49,138 @@ function login(){
                 password: passwordInput,
                 provider_name:"ziggurat"
                 })
-            }).then(response => response.json()).then(json  => {
+            }).then(response => response.json().then(json  => {
 
             try{
                 if(json.code == 200){
-
                     window.location.href = accountHome;
-                    //console.log(json);
                 }
                 else{
                     if(loginErrorMessage.classList.contains("display-none")){
-                        loginErrorMessage.classList.toggle("display-flex");
+                        loginErrorMessage.classList.toggle("display-none");
                     }
                 }
 
             }catch (error){
-                 //throw new Error(error)
                 console.log(error);
 
             }
-        })
+        }))
     }
 }
 
-function getAuthTkt(response){
-    let authTkt;
-    var json = response.json();
-
-    try{
-
-        if(json.code == 200){
+function getSessionDetails(){
+    const sessionURLFragment = "/magpie/session";
 
-            var cookieArray = response.headers.getSetCookie();
+    return fetch(sessionURLFragment,{
+        method: "GET",
+        headers: {
+            "Accept": "application/json, text/plain",
+            "Content-Type": "application/json"
+            }
+        }).then(response => { return response.json().then(json =>{
+                return json;
+            })
+        })
+}
 
-            if(cookieArray.length > 0){
-                authTkt = cookieArray[0];
-                return authTkt;
+//User functions
+//--------------
+function checkUserAuthenticated(){
+    getSessionDetails().then(json => {
+        if(json.code && json.code == 200) {
+            if (json.authenticated === false) {
+                window.location.href = loginHome;
             }
         }
-        else{
-            return null;
-        }
-
-    }
-    catch (error){
-         //throw new Error(error)
-        console.log("Error caught:")
-        console.log(error);
-
-    }
+    })
 }
 
-function getNodeRegistry() {
-    const githubURL = "{{ node_registry_url }}";
-    return fetch(githubURL).then(response => {
-        return response.json().then(json => {
+function getUserDetails(){
```

Comment:
Consider caching the username and email in a global variable or in a hidden html element so that if this function is called multiple times for a given page, you don't have to make a request each and every time.